app/expenses/new/divided-cost/[userId].tsx

```diff
@@ -0,0 +1,95 @@
+import Decimal from 'decimal.js';
+import { router, useLocalSearchParams } from 'expo-router';
+import React, { useContext, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { View } from 'react-native';
+
+import Box from '@/components/ui/box/Box';
+import CustomImage from '@/components/ui/image/CustomImage';
+import Loader from '@/components/ui/loader/Loader';
+import CostTextInput from '@/components/ui/text-input/CostTextInput';
+import WeightTextInput from '@/components/ui/text-input/WeightTextInput';
+import { ExpenseCreationContext } from '@/context/expense/ExpenseCreationContext';
+import useProfilePicture from '@/hooks/attachment/UseProfilePicture';
+import useGroupMemberDetails from '@/hooks/userdetails/UseGroupMemberDetails';
+
+export default function NewExpenseDividedCost() {
+  const { t } = useTranslation();
+  const { expenseCreation, setExpenseCreation } = useContext(ExpenseCreationContext);
+  const params = useLocalSearchParams<{ userId: string }>();
+
+  const { data: userDetails } = useGroupMemberDetails(expenseCreation.groupId, params.userId);
+  const { data: userPicture } = useProfilePicture(params.userId, userDetails?.attachmentId);
+
+  const [costString, setCostString] = useState<string>(getParticipantCostAsString());
+
+  function getParticipantCostAsString(): string {
+    return expenseCreation.expenseParticipants
+      .find((participant) => participant.participantId === params.userId)!
+      .participantCost.toString()
+      .replace('.', ',');
+  }
+
+  function setNewCost() {
+    const updatedParticipants = expenseCreation.expenseParticipants.map((participant) => {
+      if (participant.participantId === params.userId) {
+        return {
+          participantId: participant.participantId,
+          participantCost: new Decimal(costString ? costString.replace(',', '.') : 0),
```

Comment:
maybe we extract that function (replace , -> . ) to some utils file to have that in one place?